arctic_training/data/hf_instruct_source.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2025 Snowflake Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Union
+
+from pydantic import Field
+from pydantic import field_validator
+from pydantic import model_validator
+from typing_extensions import Self
+
+from arctic_training.data.hf_source import HFDataSource
+from arctic_training.data.hf_source import HFDataSourceConfig
+from arctic_training.data.utils import DatasetType
+
+
+class HFInstructDataSourceConfig(HFDataSourceConfig):
```

Comment:
Perhaps let's put the unique type term at the end of the class name?
```suggestion
class HFDataSourceConfigInstruct(HFDataSourceConfig):
```

So we could add new types like
```
class HFDataSourceConfigCausal(HFDataSourceConfig):
class HFDataSourceConfigMultiModal(HFDataSourceConfig):
...
```